### PR TITLE
[Enhancement] Support set current catalog by its name

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Add sql grammar `SET CATALOG catalog_name` for changing the current session's catalog.

Note: using this sql would reset the current database.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 35 (05.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|[436, 437]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|3|00.00%|[459, 460, 461]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[309]|
|:large_blue_circle:|com/starrocks/sql/ast/SetCatalogStmt.java|0|6|00.00%|[31, 32, 33, 36, 42, 47]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|0|7|00.00%|[798, 800, 801, 803, 804, 805, 807]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CatalogAnalyzer.java|0|4|00.00%|[109, 110, 113, 114]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|11|09.09%|[486, 1005, 1007, 1008, 1009, 1010, 1011, 1012, 1013, 1014]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

**Comment by stephen-shelby** [comment]
please add some description in the problem summary

**Comment by stephen-shelby** [comment]
add ut for set statement

File: fe/fe-core/src/test/java/com/starrocks/analysis/SetCatalogStmtTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/SetCatalogStmtTest.java:5]: move this to fe/fe-core/src/test/java/com/starrocks/sql/analyzer

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 35 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|4|7|57.14%|[801, 804, 805]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|8|11|72.73%|[1009, 1010, 1011]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CatalogAnalyzer.java|3|4|75.00%|[110]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SetCatalogStmt.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

**Comment by letian-jiang** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**